### DISCUSSION TITLE
Per-environment isolated storage?

### DISCUSSION THREAD

**Discussion by kvnwolf** [vercel/community#2742] [Help] [answered]
### Summary

Although I haven't tested this yet, I am having a question. Does Vercel's new Storage features offer per-environment isolated instances? i.e. I create a Postgres database and the data is different between my production and preview environments?

So far, I have only used it for local development, however, I suspect that when I'll deploy to production, it will use the data I am already using in my development environment.

### Steps to Reproduce

N/A

**Comment by amyegan** [comment, marked as answer]
You can set up separate databases for different environments. Just make sure the appropriate environments are checked/unchecked when connecting to your project.

<img src="https://github.com/vercel/community/assets/8247105/28e8c963-e61d-49e7-98aa-cb49856ec467" alt="project setting page showing production environment selected for database connection" />

**Reply by kvnwolf** [reply]
Understood. However, that doesn't answer my question, which was mostly related with ephemeral environments per preview deployment.

**Reply by amyegan** [reply]
Sorry, I didn't pick that up from your original question. Separate databases can be used for per-environment isolation (`development`, `preview`, `production`), but automatic per-deployment instances of the database are not currently a feature.
